### PR TITLE
Select all checkbox header enhancement

### PR DESCRIPTION
**Pull request by qili26**
## Description
A few sentences describing the overall goals of the pull request's commits.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
Bug fix:
1 select all header checkbox status is not updated if all rows are selected one by one.

New feature:
1 no indeterminate status if rows are partially selected but not all selected. 

**What is the new behavior?**
Bug fix:
1 select all header checkbox status is updated if all rows are selected one by one.

New feature:
1 indeterminate status is added.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
ezgif-4-81ad42dd00

### CODE REVIEW

File: packages/react-data-grid/src/ReactDataGrid.js [hunk 1]
```diff
@@ -103,7 +105,8 @@ class ReactDataGrid extends React.Component {
             rowKey: PropTypes.string.isRequired
           }).isRequired
         })
-      ]).isRequired
+      ]).isRequired,
+      selectedRowCounts: PropTypes.number
```
[the remark this answers is not in the document]
Comment (qili26) [reply packages/react-data-grid/src/ReactDataGrid.js]: selectBy.indexes might not be always set in React data grid. 

for example, the following usage will also be OK to trigger the row selection.
```
selectBy: {
   isSelectedKey: 'isSelected'
}
```

Hence, I want the user to pass the numbers of selected rows (they know this anyway in the wrapper level of RDG).

**Comment by ashwinputhige** [comment]
@amanmahajan7  Does this PR also fix https://github.com/adazzle/react-data-grid/issues/847 ?

**Comment by qili26** [comment]
@ashwinputhige 
no, it's not. this PR only focuses on one grid use case.

**Comment by qili26** [comment]
@malonecj @diogofcunha 
Any thoughts on the first comment that @amanmahajan7 added? I need the select all status being populated in my developing grid.

**Pull request closed without merging**